surface/include/pcl/surface/ball_pivoting_front.h

```diff
@@ -0,0 +1,238 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception, Inc.
+ *
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+  
+#ifndef PCL_BALL_PIVOTING_FRONT_H_
+#define PCL_BALL_PIVOTING_FRONT_H_
+
+namespace pcl
+{
+  namespace bpa
+  {
+    /**
+     * Front manages the edges, which ones are to be pivoted, which ones are pivoted.
+     */
+    class Front
+    {
+    public:
+      typedef std::pair<uint32_t, uint32_t> Signature;
+
+      /**
+       * This class describes the pivoting ball, which rolls on points and links the mesh.
+       * Its partial definition describes an edge of the reconstructed triangle.
+       */
+      class Edge
+      {
+      protected:
+        /** list of vertice index, should have length 2 */
+        std::vector<uint32_t> id_vertices_;
+        /** index of the opposite vertice, it and id_vertices_ form the last triangle */
+        uint32_t id_opposite_;
+        /** center point of the last ball, it may not be one in the cloud */
+        Eigen::Vector3f center_;
+        /** whether this ball it to the it rolled in the back face of surface (not normal direction) */
+        bool is_back_ball_;
+    
+      public:
+        Edge ();
+    
+        /**
+         * a fake constructor with only vertice index on edge
```

Comment:
I'm not sure if "fake" is a good terminology, it's not clear what is implied. What is the state of the other variables? Is an edge constructed this way valid?